Assets/Modules/Health/PlayerLifeController.cs

```diff
@@ -44,15 +46,15 @@ private void OnDestroy()
     void Start()
     {
         crackIndex = 0;
-        isAlive = true;
+        PlayerManager.Instance.isAlive = true;
```

Comment:
Oh this is not good. We shouldn't be able to set the value of ` PlayerManager.Instance.isAlive` we should only read from it. So I suggest doing the following thing. Do not use PlayerManager in player scripts in general and make the isAlive variable a property like this in PlayerManager.cs:

```
[Serializefield] private  PlayerLifeController playerLifeController;

public bool IsAlive => playerLifeController.isAlive;
```
This way you are protecting this super accessible variable from getting wrongly set by other classes.

Reply:
About this one I do have a question. Then how do we change the value inside the PlayerManager? Should I subscribe inside PlayerManager to the "OnPlayerDiedEvent" to change the value of this variable within this script?